src/utils/react-query-factory.js

```diff
@@ -0,0 +1,93 @@
+import { fetchCourseMetadata } from '../components/app/routes/queries/courseMetadata';
+import { fetchUserEntitlements } from '../components/app/routes/queries/userEntitlements';
+import { fetchCanRedeem } from '../components/app/routes/queries/canRedeemCourse';
+import { fetchEnterpriseLearnerData } from '../components/app/routes/queries/enterpriseLearner';
+import { fetchEnterpriseCourseEnrollments } from '../components/app/routes/queries/enterpriseCourseEnrollments';
+import { fetchBrowseAndRequestConfiguration } from '../components/app/routes/queries/subsidies/browseAndRequest';
+import { fetchRedeemablePolicies } from '../components/app/routes/queries/subsidies/policies';
+import { fetchSubscriptions } from '../components/app/routes/queries/subsidies/subscriptions';
+import { fetchCouponCodes } from '../components/app/routes/queries/subsidies/couponCodes';
+import { fetchEnterpriseOffers } from '../components/app/routes/queries/subsidies/enterpriseOffers';
+import { fetchEnterpriseCuration } from '../components/app/routes/queries/contentHighlights';
+
+// The keys that map to enterpriseQueryKeys that is used to define specific queries
+// and used to specifically invalidate queries
+const enterprise = createQueryKeyStore('enterprise', {
+  enterpriseCustomer: (enterpriseUuid) => ({
+    queryKey: [enterpriseUuid],
+    contextQueries: {
+      course: (courseKey) => ({
+        queryKey: [courseKey],
+        contextQueries: {
+          contentMetadata: {
+            queryKey: null,
+            queryFn: async ({ queryKey }) => fetchCourseMetadata(queryKey[2], queryKey[4]),
+          },
+          canRedeem: (availableCourseRunKeys) => ({
+            queryKey: [availableCourseRunKeys],
+            queryFn: async ({ queryKey }) => fetchCanRedeem(queryKey[2], availableCourseRunKeys),
+          }),
+        },
+      }),
+      user: {
+        queryKey: ['entitlements'],
+        queryFn: async () => fetchUserEntitlements(),
+      },
+      enrollments: {
+        queryKey: null,
+        queryFn: async ({ queryKey }) => fetchEnterpriseCourseEnrollments(queryKey[2]),
+      },
+      contentHighlights: {
+        queryKey: null,
+        contextQueries: {
+          configuration: {
+            queryKey: null,
+            queryFn: async ({ queryKey }) => fetchEnterpriseCuration(queryKey[2]),
+          },
+        },
+      },
+    },
+  }),
+  enterpriseLearner: (username, enterpriseSlug) => ({
+    queryKey: [username, enterpriseSlug],
+    queryFn: async () => fetchEnterpriseLearnerData(username, enterpriseSlug),
+  }),
+  userSubsidy: (enterpriseUuid) => ({
+    queryKey: [enterpriseUuid],
+    contextQueries: {
+      browseAndRequest: {
+        queryKey: null,
+        contextQueries: {
+          configuration: (userEmail) => ({
+            queryKey: [userEmail],
+            queryFn: async ({ queryKey }) => fetchBrowseAndRequestConfiguration(queryKey[2], userEmail),
+          }),
+        },
```

Comment:
[curious/suggestion] Should we also configure the query keys for `license-requests` and `coupon-code-requests` API queries? Currently, we call these 2 additional endpoints within `fetchBrowseAndRequestConfiguration` but perhaps that's not the best approach given there's no specific query key for `license-requests` vs. `coupon-code-requests`. For example, if we were to invalidate the `browseAndRequest._ctx.configuration` query key, it will currently also invalidate the queries around retrieving the actual requests (in addition to the configuration). Might imagine we'd want to invalidate query cache for just license requests, for example?

Also worth nothing, it appears all 3 queries under `browseAndRequest` currently rely on a `userEmail`. Given this, I wonder if it'd make sense to pass the `userEmail` as a function arg to the top-level `browseAndRequest` query key grouping rather than to the individual contextual queries? That way, the `userEmail` would be accessible via the `queryKey` array to all nested/contextual queries by default.